app/models/talk.rb

```diff
@@ -41,26 +52,56 @@ class Talk < ApplicationRecord
   # delegates
   delegate :name, to: :event, prefix: true, allow_nil: true
 
+  before_save :compute_embedding, if: :must_compute_embedding?
+
   # search
-  meilisearch do
+  meilisearch primary_key: :id, enqueue: true, raise_on_failure: Rails.env.development? do
     attribute :title
     attribute :description
     attribute :slug
-    attribute :video_id
+    # ⚠️ This `video_id` attribute makes indexing (silently) fail with v1.3.2 of meilisearch. Error message from meilisearch (GET /tasks):
+    #   "The primary key inference failed as the engine found 2 fields ending with `id` in their names: 'id' and 'video_id'. Please specify the primary key manually using the `primaryKey` query parameter"
+    # Adding a custom primary_key: :id above didn't make any difference, so removing this attribute for now.
+    # attribute :video_id
     attribute :video_provider
     attribute :thumbnail_sm
     attribute :thumbnail_md
     attribute :thumbnail_lg
     attribute :speakers do
       speakers.pluck(:name)
     end
+    # ⚠️ This must return nil and not an empty array if no vector is available.
+    # Otherwise all other indexing tasks with non-zero vector arrays will silently fail, since the engine will expect all vectors to have the same length.
```

Comment:
@Kerollmops this was by far the most time-consuming issue I have with meilisearch vector indexing: seems like meilisearch rejects any document that doesn't have a vector of the same size as the first one it has seen. If you mistakenly send a zero-sized vector (or incorrectly sized vector) for your first document, then all other documents with a different vector size will be rejected. This should be put in a big warning somewhere, or hopefully have some better error message in the server logs (had to inspect the tasks list to see this error).

Reply:
Thank you for the report @crohr. I shared your feedback with the team and will see what we can do. We [recently fixed a bug with the `_vectors` field](https://github.com/meilisearch/meilisearch/pull/4039) that was producing issues when specifying multiple vectors for a single document. This one will be released in v1.3.2.